### PR TITLE
reduce multicluster testing topology to 3 clusters

### PR DESCRIPTION
**Pull request by stevenctl**
As discussed in T&R, reducing the topology of multi-cluster tests to be 3 clusters on two networks.

Cluster 1: Primary Cluster
Cluster 2: Remote Cluster, uses cluster 1 as primary
Cluster 3: Primary Cluster

All Primary clusters can discover services/endpoints via remote secrets in every other cluster in the mesh. 

We can add `--topology-config /prow/config/topology/multicluster-large.json` for the postsubmit versions of the test if we want to use the old 5 cluster topology there, or locally. This flag and others need to be documented as part of #29610

### CODE REVIEW

**Comment by howardjohn** [comment]
> Cluster 1: Primary Cluster
> Cluster 2: Primary Cluster

Do both of these clusters have kubeconfigs for 1, 2, and 3?

**Comment by stevenctl** [comment]
@howardjohn, yes. Updated the description to capture this, updating test framework docs as well.

**Comment by howardjohn** [comment]
Looks good but some tests are failing

**Comment by stevenctl** [comment]
The test framework VM bootstrap code probably needs some logic to point at the control plane cluster even if the "vm" pod is in a remote cluster

File: pkg/test/framework/components/echo/kube/instance.go [hunk 1]
```diff
@@ -289,6 +299,18 @@ func createVMConfig(ctx resource.Context, c *instance, cfg echo.Config) error {
 	return nil
 }
 
+func customizeWorkloadGroup(cfg echo.Config, wg []byte) ([]byte, error) {
```
Comment (stevenctl) [inline pkg/test/framework/components/echo/kube/instance.go:302]: not needed yet.. will be if we turn auto-register on in multi-network tests

**Review by howardjohn — APPROVED** [review]

**Comment by stevenctl** [comment]
The VM test was failing because the VM used cluster 0 control plane, but was running in cluster 2. The `CLUSTER_ID` meta on the VM's proxy was `cluster-0`, that's what we'd do with real VMs, but in reality this "vm" was running in cluster-2, which likely caused some issues. I've changed the topology so we have 1 primary per-network. I don't think we gain anything by testing "a remote on a different network" and without the primary on the other network, we can't effectively test multi-network VMs.

Why the security test was failing: we started deploying headless services to non-primary clusters. This is fine in our networking tests, but breaks in security tests. Maybe @jtrbs can provide more info? 


cc @nmittler and @howardjohn since the new reduced topology is changed.

**Comment by stevenctl** [comment]
Hold is there so I can fix mc telemetry test

**Review by esnible — APPROVED** [review]
LGTM

**Review by morvencao — APPROVED** [review]

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**